src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -93,23 +150,49 @@ public static async Task OrasExecutionAsync(string command, string orasWorkingDi
             }
             catch (Exception ex)
             {
-                throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, "Oras process failed", ex);
+                throw new OCIClientException(TemplateManagementErrorCode.OrasProcessFailed, "Oras process failed", ex);
             }
 
             StreamReader errStreamReader = process.StandardError;
+            StreamReader outputStreamReader = process.StandardOutput;
             await Task.WhenAny(eventHandled.Task, Task.Delay(Constants.TimeOutMilliseconds));
             if (process.HasExited)
             {
                 string error = errStreamReader.ReadToEnd();
-                if (!string.IsNullOrEmpty(error))
+                if (!string.IsNullOrEmpty(error) || process.ExitCode != 0)
                 {
-                    throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, error);
+                    throw new OCIClientException(TemplateManagementErrorCode.OrasProcessFailed, "Oras process failed." + error);
                 }
+
+                return outputStreamReader.ReadToEnd();
             }
             else
             {
-                throw new OrasException(TemplateManagementErrorCode.OrasTimeOut, "Oras request timeout");
+                try
+                {
+                    process.Kill();
+                }
+                catch (Exception ex)
+                {
+                    throw new OCIClientException(TemplateManagementErrorCode.OrasTimeOut, "Oras request timeout. Fail to kill oras process.", ex);
+                }
+
+                throw new OCIClientException(TemplateManagementErrorCode.OrasTimeOut, "Oras request timeout");
+            }
+        }
+
+        private void InitClientEnvironment()
+        {
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.OrasCacheEnvironmentVariableName)))
+            {
+                Environment.SetEnvironmentVariable(Constants.OrasCacheEnvironmentVariableName, Constants.DefaultOrasCacheEnvironmentVariable);
             }
         }
+
+        private string LoadManifestContentFromCache(string digest)
```

Comment:
Does this mean that we require customers to set ORAS_CACHE in the future?

Reply:
Yes, Oras will save manifest in cache, so we need to enable cache environment for Oras. Customers can set ORAS_CACHE , otherwise we set the environment by default.